components/brave_origin/ad_block_only_mode_policy_manager.cc

```diff
@@ -0,0 +1,86 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_origin/ad_block_only_mode_policy_manager.h"
+
+#include "base/feature_list.h"
+#include "base/no_destructor.h"
+#include "brave/components/brave_shields/core/common/features.h"
+#include "brave/components/brave_shields/core/common/pref_names.h"
+#include "components/prefs/pref_service.h"
+
+namespace brave_origin {
+
+// static
+AdBlockOnlyModePolicyManager* AdBlockOnlyModePolicyManager::GetInstance() {
```

Comment:
why is this in brave_origin?

Reply:
this seems like it should be in brave_shields?

Reply:
I did it after discussion with @bbondy that BraveOriginPolicyManager will be generalized to BravePolicyManager. And AdBlockOnlyModePolicyManager can stay here until it is done. There is a ticket for generalization: https://github.com/brave/brave-browser/issues/50077